File: trunk/PolarMath/src/org/macondo/polar/evaluation/spectrum/ULF.java
package org.macondo.polar.evaluation.spectrum;

import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;
import org.macondo.polar.util.FFT;
import org.macondo.polar.util.Periodogram;
import org.macondo.polar.util.Square;

public class ULF implements Evaluation<Double>{
	public Double evaluate(Training training) {
		List<Periodogram> periodogram = training.evaluate(new FFT());
		return new Square(periodogram, 0, 0.0033).Calculate();
	}
}

File: trunk/MetroApp/HrmMath/Evaluation/Spectrum/ULF.cs
using HrmMath.Data;
using HrmMath.Util;
using System.Collections.Generic;

namespace HrmMath.Evaluation.Spectrum
{
    public class ULF : IEvaluation<double>
    {
        public double Evaluate(SessionData training)
        {
            var periodogram = training.Evaluate( new Lomb() );

            return new Square( periodogram, 0, 0.0033 ).Calculate();
        }
    }
}
